src/test/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessorTests.java

```diff
@@ -114,6 +119,28 @@ public void testInferenceSentences_whenExceptionFromMLClient_thenFailure() {
         Mockito.verifyNoMoreInteractions(resultListener);
     }
 
+    public void testInferenceSentences_whenNodeNotConnectedException_thenRetry() {
```

Comment:
Add these test cases also:
1. Test case which tests when all the retries are finished.
2. Test cases which test when these retryable exceptions are not thrown we should get a failure.

Reply:
Two cases covered:
1. check the method `testInferenceSentences_whenNodeNotConnectedException_thenRetry_3Times` and the `Mockito.verify(client, times(4))` stands for the client method has been invoked for 4 times in total.
2. Add a new method `testInferenceSentences_whenNotConnectionException_thenNoRetry` and the verify has time of 1, meaning method only run once.